internal/maps/maps.go

```diff
@@ -0,0 +1,100 @@
+package maps
+
+import (
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"os"
+	"strings"
+)
+
+// FromSlice converts a slice of dot-delimited string values into a map[string]any.
+// For example:
+// "a.b.c=123","a.b.d=124" would return { "a": { "b": { "c": 123, "d": 124 } } }
+func FromSlice(slice []string) map[string]any {
+	m := map[string]any{}
+
+	for _, s := range slice {
+		// s is going to be in the format of
+		// a.b.c=xyz
+		parts := strings.SplitN(s, "=", 2)
```

Comment:
Is this going to cause any issues if the value contains `=` as in a base64 encoded string (e.g. `DEADBEEF==`)?

Reply:
[No](https://go.dev/play/p/bFgJ4kNvAAl), this is why I am using `SplitN` with `2`, so it will only split on the first `=`.

If the key was base64 encoded, then that would be a problem.